### ISSUE TITLE
implement IPv6

### ISSUE THREAD

**Issue by nis65** [nis65/ansible-debian-router#60]
as mentioned in the main `README.md`

**Comment by nis65** [comment]
Started to investigate / test, see #63.

IPv6 prefix delegation with `ifupdown` does not allow to use all configuration options available to `dhclient`. Especially I did not find a way to set `--prefix-len-hint`.

The command line to call `dhclient` is hardcoded in `ifupdown`, e.g. [here](https://sources.debian.org/src/ifupdown/0.8.41/inet6.defn/#L161).

I also found this [workaround](https://stackoverflow.com/questions/71905785/ifupdown-change-parameters-dhclient), i.e. add some `pre-up` statements to mimic the behaviour of `ifupdown` calling dhclient.

Maybe I should investigate into [ifupdown2](https://sources.debian.org/src/ifupdown2/3.0.0-1/)?

Or fall back to [wide-dhcpv6](https://sources.debian.org/src/wide-dhcpv6/20080615-23/)?

**Comment by nis65** [comment]
Well, it's not that easy. Both `wide-dhcpv6` and `dibbler` (another ipv6 dhcp client implementation) seem not to be actively maintained any more. So it makes sense to stick to `dhclient`.

Before investing more time, I tested the `--prefix-len-hint` manually:

Request: 
```
dhclient -6 -v -pf /run/dhclient6.enp1s0.pid -lf /var/lib/dhcp/dhclient6.enp1s0.leases -I -P -N --prefix-len-hint 56 -df /var/lib/dhcp/dhclient.enp1s0.leases enp1s0
```
Release
```
dhclient -6 -v -r -pf /run/dhclient6.enp1s0.pid -lf /var/lib/dhcp/dhclient6.enp1s0.leases -I -df /var/lib/dhcp/dhclient.enp1s0.leases enp1s0
```

And it turns out that my provider does not honour the `prefix-len-hint`, regardless of my setting (longer/shorter than 60), I always get a `/60`:

~~~
dhclient -6 -v -pf /run/dhclient6.enp1s0.pid -lf /var/lib/dhcp/dhclient6.enp1s0.leases -I -P -N --prefix-len-hint 64 -df /var/lib/dhcp/dhclient.enp1s0.leases enp1s0
...
XMT:  | | X-- Request prefix ::/64.
...
RCV:  | | X-- IAPREFIX 2001:xxxx:xxxx:xxxx::/60
...
~~~

So I'll stick to `dhclient` without supporting the `prefix-len-hint`. I could not make use of it anyway. :-(

**Comment by nis65** [comment]
Current state / bugs:
* ~~upv6if needs also an option that works when prefix delegation is not available. E.g. in my lab where I want to test some IPv6 stuff~~. The current template already allows to write the IPv6 config in a generic way. My test router already got a routable IPv6 via SLAAC, need to test!
* ~~there must be a way to implement the nft rules IPv4/v6 agnostic!~~ (fixed with #67)
* ~~check afraid dns config for ipv6~~: Afraid does not support IPv6 dynamic dns updates currently.
* openvpn (with nft integration) ~~is~~ partially missing: individual ipv6 nft rules not implemented yet.

**Comment by nis65** [comment]
see #69 too and #70

**Comment by nis65** [comment]
And resolved with #71
